qiskit/quantum_info/operators/symplectic/clifford_circuits.py

```diff
@@ -179,12 +179,15 @@ def _append_h(clifford, qubit):
     Returns:
         Clifford: the updated Clifford.
     """
-    x = clifford.table.X[:, qubit]
-    z = clifford.table.Z[:, qubit]
-    clifford.table.phase ^= x & z
+    x = clifford.paulis.x[:, qubit]
+    z = clifford.paulis.z[:, qubit]
+
+    count_y = clifford.paulis._count_y()
+    clifford.paulis.phase += (x & z).astype(int) * 2
     tmp = x.copy()
     x[:] = z
     z[:] = tmp
+    clifford.paulis.phase += clifford.paulis._count_y() - count_y
```

Comment:
This stuff with `_count_y` doesn't seem very efficient, and ideally we shouldn't be accessing an internal method of another class.  Isn't the number of `y` elements going to be the same in both lines?  There's a Y whenever there's an X and a Z in the same location, and swapping the X and Z arrays won't change those locations.

Reply:
Yes, this is exactly the difference between a PauliTable and a PauliList, and the performance will inevitably decrease because of tracking phases when reimplementing with a PauliList. I think this performance regression was discussed last year once accepted, but I hope it will be improved in the future (for example, by using Rust).